File: Easy/3658_GCD_of_Odd_and_Even_Sums/Solution.go
package main

import "fmt"

// Past the function from leetcode here
func gcdOfOddEvenSums(n int) int {
	var oddSum int = 0
	var evenSum int = 0

	for i := 1; i <= n*2; i += 2 {
		oddSum += i
	}

	for i := 2; i <= n*2; i += 2 {
		evenSum += i
	}

	for evenSum != 0 {
		oddSum, evenSum = evenSum, oddSum%evenSum
	}

	return oddSum
}

func main() {
	// test cases 1
	fmt.Println(gcdOfOddEvenSums(4))

	// test cases 2
	fmt.Println(gcdOfOddEvenSums(5))

}

File: Easy/3658_GCD_of_Odd_and_Even_Sums/Solution.py
from typing import List, Optional


class Solution:
    def gcdOfOddEvenSums(self, n: int) -> int:
        oddSum = 0
        evenSum = 0

        for i in range(1, n * 2 + 1, 2):
            oddSum += i

        for i in range(2, n * 2 + 1, 2):
            evenSum += i

        while evenSum != 0:
            oddSum, evenSum = evenSum, oddSum % evenSum

        return oddSum


if __name__ == "__main__":
    solution = Solution()

    # test cases 1
    print(solution.gcdOfOddEvenSums(4))
    
    # test cases 2
    print(solution.gcdOfOddEvenSums(5))